packages/react-native/Libraries/Network/RCTFileRequestHandler.mm

```diff
@@ -25,8 +25,14 @@ @implementation RCTFileRequestHandler {
 
 - (void)invalidate
 {
-  [_fileQueue cancelAllOperations];
-  _fileQueue = nil;
+  if (_fileQueue) {
+    for (NSOperation *operation in _fileQueue.operations) {
+      if ([operation isKindOfClass:[NSOperation class]] && !operation.isCancelled && !operation.isFinished) {
+        [operation cancel];
+      }
+    }
+    _fileQueue = nil;
+  }
```

Comment:
This is not necessary. You can only add `NSOperation` to an `NSOperationQueue` and `cancellAllOperations` runs the same code you are manually writing.

Reply:
This is the official Apple docs for [`cancelAllOperations`](https://developer.apple.com/documentation/foundation/nsoperationqueue/1417849-cancelalloperations).

> This method calls the [cancel](https://developer.apple.com/documentation/foundation/nsoperation/1411672-cancel) method on all operations currently in the queue.
> Canceling the operations does not automatically remove them from the queue or stop those that are currently executing. For operations that are queued and waiting execution, the queue must still attempt to execute the operation before recognizing that it is canceled and moving it to the finished state. For operations that are already executing, the operation object itself must check for cancellation and stop what it is doing so that it can move to the finished state. In both cases, a finished (or canceled) operation is still given a chance to execute its completion block before it is removed from the queue.

And this is the docs of [`NSOperation cancel`](https://developer.apple.com/documentation/foundation/nsoperation/1411672-cancel).

In any case, calling `cancel` on an already cancelled or finished  operation does not crash the app.

I believe that the crash is happening inside one of the operations that are being cancelled and that's why the crash reporter reports the crash there.

Reply:
Is it possible that even though we already put assurance for the corresponding code to make it only executed on the main thread or another sole thread, e.g. the JS thread, but because of the nature of object pointer reference, the operation object could still be shared among multiple thread contexts, then the situation you said above happens.